### ISSUE TITLE
dynamic group by for list views

### ISSUE THREAD

**Issue by christianhugoch** [saltcorn/saltcorn#2111]
**Is your feature request related to a problem? Please describe.**
For data like in this screenshot:

![dynamic_group_by_example](https://github.com/saltcorn/saltcorn/assets/10777655/76300a4c-27db-448a-845b-e2edd7ccbe59)
I would like to group by "Author" so that I can sum up all pages of unique authors.

I believe we have no option for dynamic SQL GROUP BYs, or does anyone have an idea?

**Describe the solution you'd like**

Perhaps we could add group by options in the configuration steps of list views.

**Comment by glutamate** [comment]
If Author is a key to a different table, you can (in this case) make a List on that table and add an aggregation.

Also, the pivottable module is very powerful in this respect

**Comment by christianhugoch** [comment]
Thanks, I'll take a look into pivottable

**Comment by Ingenu1ty** [comment]
> Also, the pivottable module is very powerful in this respect

Yes, of course, pivottable is a very cool plugin in terms of aggregation, but even it can be expanded with **grouping capabilities**. 

I mean [this wonderful fork](https://github.com/jjagielka/pivottable-grouping)! I think it wouldn't take much work to adapt an [existing SC plugin](https://github.com/saltcorn/pivottable) to this fork... So it would be great if there were two SC pivottable plugins (so as not to break backward compatibility in existing SC projects*): the original one and the fork of it with grouping capabilities. 

*Or maybe there is no threat to backward compatibility and it’s easier to have one plugin, but with the settings... I really don’t know which way would be better...

**Comment by pyhedgehog** [comment]
ListSQL plugin has such ability. :-)

**Comment by christianhugoch** [comment]
Can you send me a link to that plugin?

**Comment by pyhedgehog** [comment]
> Can you send me a link to that plugin?
https://github.com/saltcorn/sql-list
